### PR TITLE
server: get/put cluster meta

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut

### CODE REVIEW

File: server/cluster.go [hunk 1]
```diff
@@ -574,3 +574,40 @@ func (c *raftCluster) PutStore(store *metapb.Store) error {
 
 	return nil
 }
+
+func (c *raftCluster) GetMeta() (*metapb.Cluster, error) {
```
Comment (ngaut) [inline server/cluster.go]: Do we need to clone meta ? Looks like there will be race if there are multiple reader and writer.

Comment (siddontang) [inline server/cluster.go]: we clone it below, see line 583.

Comment (ngaut) [inline server/cluster.go]: Gotcha.

**Comment by ngaut** [comment]
LGTM

**Pull request merged**